trucks.js

```diff
@@ -1,511 +1,597 @@
+      type: ['Sweets'],
+      payment: ['Cash', 'Cards'],
+      description: 'Handmade ice creams!',
+      website: 'http://www.fulltilticecream.com',
+      Facebook: 'https://www.facebook.com/fulltilticecreams',
+      Twitter: 'https://twitter.com/fticecream'
+  }
+];
+
+// this module should support the following methods:
+// getTrucks() - return all trucks
+// getTruck(name) - return the truck object matching 'name'
+// getFoodTypes() - return unique list of all associated food types (underscore has a function to help)
+// filterByDay(day) - return trucks with 'day' in schedule (use your filterByDay function from Module 3 homework)
+// filterByFoodType(foodType) - return trucks with associated 'foodType'
+
+//Extend the trucks module, trucks.js provided with the following methods:
+//getTrucks This method takes no parameters and returns a list of all the truck objects in the module.
+
+function getTrucks() {
+    return foodTrucks;
+}
+
+var getTrucks = function () {
+    var truckList = [];
+    for (var i = 0; i < foodTrucks.length; i++) {
+        truckList.push(foodTrucks[i].name);
+    }
+
+    return truckList;
+};
+
+//getTruck(name) This method takes one parameter, a string representing the name of a food truck. The function returns the truck object matching the name provided. 
+
+var getTruck = function (name) {
+    for (var i = 0; i < foodTrucks.length; i++) {
+        if (name === foodTrucks[i].name) {
+            return foodTrucks[i];
+        } else {
+            return 'Sorry No truck';
+        }
+    }
+};
+
+// getFoodTypes This method takes no parameters and returns a list of all the food types offered by the various trucks in the module. 
+var getFoodTypes = function () {
+    var foodType = [];
+    for (var i = 0; i < foodTrucks.length; i++) {
+        var typeOfFood = foodTrucks[i].type.toString();
+        foodType.push(typeOfFood);
+    }
+
+    return foodType;
+};
+
+
+// filterTrucksByDay(day) This method takes one parameter, a string representing a day of the week. The function returns the list of trucks open on the day provided.
+var filterByDay = function (day) {
+    var truckList = [];
+    for (var i = 0; i < foodTrucks.length; i++) {
+        if (foodTrucks[i].schedule) {
+            for (var j = 0; j < foodTrucks[i].schedule.length; j++) {
+                var dayOfTruck = foodTrucks[i].schedule[j];
+                if (dayOfTruck === day) {
+
+                    truckList.push(foodTrucks[i].name);
+                }
+            }
+        }
+    }
+
+    return truckList;
+};
+
+// filterTrucksByFoodType(foodType) This method takes one parameter, a string representing a food type. The function returns the list of trucks that serve the provided type of food. 
+var filterByFoodType = function (foodType) {
+    var typeList = [];
+    for (var i = 0; i < foodTrucks.length; i++) {
+        var typeOfFood = foodTrucks[i].type.toString();
+        if (foodType === typeOfFood) {

```

Comment:
`type` is an array, and by calling toString on the entire array, you end up creating something that won't compare to the `foodType` value passed in to the method properly.